src/workerd/api/tests/transform-streams-test.js

```diff
@@ -0,0 +1,437 @@
+
+    const { readable, writable } = transform;
+
+    const writer = writable.getWriter();
+
+    const res = await Promise.allSettled([
+      writer.write('hello'),
+      writer.write('there'),
+      writer.close(),
+      consume(readable),
+    ]);
+
+    // The two writes will succeed.
+    strictEqual(res[0].status, 'fulfilled');
+    strictEqual(res[1].status, 'fulfilled');
+
+    // The close and the consume will reject.
+    strictEqual(res[2].status, 'rejected');
+    strictEqual(res[3].status, 'rejected');
+    strictEqual(res[2].reason.message, 'boom');
+    strictEqual(res[3].reason.message, 'boom');
+  },
+};
+
+// Test async error during flush
+export const asyncErrorDuringFlush = {
+  async test() {
+    const transform = new TransformStream({
+      async flush() {
+        await scheduler.wait(1);
+        throw new Error('boom');
+      },
+    });
+    ok(!(transform instanceof IdentityTransformStream));
+
+    const { readable, writable } = transform;
+
+    const writer = writable.getWriter();
+
+    const res = await Promise.allSettled([
+      writer.write('hello'),
+      writer.write('there'),
+      writer.close(),
+      consume(readable),
+    ]);
+
+    // The two writes will succeed.
+    strictEqual(res[0].status, 'fulfilled');
+    strictEqual(res[1].status, 'fulfilled');
+
+    // The close and the consume will reject.
+    strictEqual(res[2].status, 'rejected');
+    strictEqual(res[3].status, 'rejected');
+    strictEqual(res[2].reason.message, 'boom');
+    strictEqual(res[3].reason.message, 'boom');
+  },
+};
+
+// Test write backpressure
+export const writeBackpressure = {
+  async test() {
+    let expectedReadSize = 2;
+    const transform = new TransformStream(
+      {
+        transform(chunk, controller) {
+          strictEqual(controller.desiredSize, expectedReadSize--);
+          controller.enqueue(chunk);
+        },
+      },
+      { highWaterMark: 2 },
+      { highWaterMark: 2 }
+    );
+
+    const writer = transform.writable.getWriter();
+    strictEqual(writer.desiredSize, 2);
+
+    const promises = [writer.write('hello'), writer.write('there')];
+
+    strictEqual(writer.desiredSize, 0);
+
+    await Promise.allSettled(promises);
+
+    strictEqual(writer.desiredSize, 2);
+  },
+};
+
+// Test TransformStream cancel function
+export const transformStreamCancel = {
+  async test() {
+    // Verify that a TransformStream's cancel function is called when the
+    // readable is canceled or the writable is aborted. Verify also that
+    // errors thrown by the cancel function are propagated.
+    {
+      const cancelFn = mock.fn(async (reason) => {
+        strictEqual(reason, 'boom');
+        await scheduler.wait(10);
+      });
+      const { readable } = new TransformStream({
+        cancel: cancelFn,
+      });
+      strictEqual(cancelFn.mock.callCount(), 0);
+      await readable.cancel('boom');
+      strictEqual(cancelFn.mock.callCount(), 1);
```

Comment:
`callCount` is bumped before `scheduler.wait` though right? So in theory the test changed?